src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -417,23 +419,61 @@ public async Task<JObject> Resolve(InvocationExpressionSyntax method, Dictionary
                     using var commandParamsObjWriter = new MonoBinaryWriter();
                     if (isTryingLinq == 0)
                         commandParamsObjWriter.WriteObj(objectId, context.SdbAgent);
+
                     if (method.ArgumentList != null)
                     {
-                        commandParamsObjWriter.Write((int)method.ArgumentList.Arguments.Count + isTryingLinq);
+                        int passedArgsCnt = method.ArgumentList.Arguments.Count;
+                        int sentToDebuggerParamsCnt = passedArgsCnt;
+                        var methodParamsInfo = new List<ParameterInfo>();
+                        var methodInfo = await context.SdbAgent.GetMethodInfo(methodId, token);
+                        if (methodInfo != null) //null for e.g. ToList()
+                        {
+                            methodParamsInfo = methodInfo.Info.ParametersInfo.ToList();
+                            int methodParamsCnt = methodParamsInfo.Count;
+                            if (passedArgsCnt > methodParamsCnt)
+                                throw new ReturnAsErrorException($"Unable to evaluate method '{methodName}. Too many arguments passed.", "ArgumentError");
+                            int optionalParamsCnt = methodInfo.Info.OptionalParametersCnt;
+                            if (passedArgsCnt + optionalParamsCnt >= methodParamsCnt)
+                                sentToDebuggerParamsCnt = methodParamsCnt;
```

Comment:
`sentToDebuggerParamsCnt` should always be equal to `methodParamsCnt`. CLR doesn't really support optional parameters. It's a C# language level feature. If you call a method with some optional args missing, then the compiler use the default param values for them. So, the method is always called with all the args.

@thaystg I'm assuming that this holds true when using the debug api too.